updated index page and created and images folder (#460)
kubernetes-sigs/service-apis: docs-src/index.md

# Introduction

Service APIs is an open source project, working under

> [SIG-NETWORK][sig-network], that is evolving service networking APIs within the Kubernetes ecosystem. Service APIs are the APIs you use to expose your Kubernetes applications - Services, Ingress, and more. The Gateway API is the primary API being developed to unify service networking under a core set of networking APIs.

Reviewer: Naming is a bit confusing here. I think Service APIs covers GatewayClass, Gateway, *Route, and BackendPolicy resources. This represents an evolution of the Ingress API and takes on part of the functionality from the Service API, especially where `type=LoadBalancer`. I think it might be simplest to remove the last couple sentences in this paragraph, but maybe there's a better option.
```suggestion
[SIG-NETWORK][sig-network], that is evolving service networking APIs within the Kubernetes ecosystem.
```

> ### What is the Gateway API?

Reviewer: I don't think we should use `Gateway API` as a name here. Service APIs are broader than that resource.

Reviewer 2: +1 here.
I found this term being used GCP's repo as well: https://github.com/GoogleCloudPlatform/gke-networking-recipes/tree/master/gateway

There is already some confusion brewing: https://github.com/kubernetes-sigs/service-apis/issues/332#issuecomment-719401062

> The Gateway API is an effort by a consortioum of companies and users to evolve the current Kubernetes Ingress API. Gateway aims to improve upon Ingress by becoming a more expressive, extensible, role-oriented API while remaining a generic routing API that has many implementations and broad industry support. 

Reviewer 2: All of our existing text in markdown files is approximately 80 column long. Could you please wrap lines in this patch accordingly?

Reviewer: Good catch - if you happen to be using VSCode I highly recommend [Rewrap](https://marketplace.visualstudio.com/items?itemName=stkb.rewrap) to do this automatically.

Author: rewrapped to 80 char

Gateway is really a collection of API resources - `GatewayClass`, `Gateway`, `HTTPRoute`, `TCPRoute`, and so on. These resources are used together to model a wide variety of different networking use-cases. See [API Concepts](concepts.md) to learn more.

> ![Gateway API Model](./images/api-model.png)

Reviewer: Similar comments around naming here, I think we should be using "Service APIs" wherever possible for consistency.

Author: I changed it to using "Service APIs" in a more generic way. It does feel strange to not have a more concrete name for the project though because "Service APIs" feels like such an amorphous reference for what we're talking about, but that's a discussion for another day :)

> How does Gateway improve upon current standards like Ingress?

Reviewer: ```suggestion
How do these APIs improve upon current standards like Ingress?
```

- **Cross-Namespace references** - Routes across different Namespaces can bind to Gateways. This allows for shared networking infrastructure despite Namespace segmentation for workloads.
- **Classes** - GatewayClasses formalize types of load balancing implementations. These classes make it easy and explicit for users to understand what kind of capabilities are available as a resource model itself.

[sig-network]: https://github.com/kubernetes/community/tree/master/sig-network

### Where to get started

To get started, please read through [API concepts](concepts.md). These documents give the necessary background
to understand the API and the use-cases it targets. Once you have a good understanding of the API at a higher-level, please follow one of our [guides](guides.md) to dive deeper into different parts of the API.

For a complete API reference, please refer to:

- [API reference](spec.md)
- [Go docs for the package](https://pkg.go.dev/sigs.k8s.io/service-apis/apis/v1alpha1)

### How to get involved

> Gateway has many contributors and we welcome anybody and everybody to get involved. Join our weekly meetings, file issues, or ask questions in slack. No contribution is too small - even opinions matter! 

Reviewer: ```suggestion
This project has many contributors and we welcome anybody and everybody to get involved. Join our weekly meetings, file issues, or ask questions in Slack. No contribution is too small - even opinions matter! 
```
